### PR TITLE
Introduce support for alias attributes

### PR DESCRIPTION
**Pull request by leonidkroka**
Probably it's related to issues:
https://github.com/zdennis/activerecord-import/issues/728:
```
class MyModel < ActiveRecord::Base
  alias_attribute :foo, :baz
end

MyModel.import(%i[foo bar], %w[value_foo, value_baz], validate: false, on_duplicate_key_update: { conflict_target: %i[bar], columns: %i[foo] })
```

https://github.com/zdennis/activerecord-import/issues/795:
```
class MyModel < ActiveRecord::Base
  self.ignored_columns = [:bar]
end

MyModel.import(%i[foo bar], %w[value_foo, value_baz], validate: false, on_duplicate_key_ignore: true)
MyModel.import(%i[foo bar], %w[value_foo, value_baz], validate: false, on_duplicate_key_update: { conflict_target: %i[foo], columns: %i[bar] })
```

### CODE REVIEW

**Comment by jkowens** [comment]
Awesome, thank you for this update! 🎉

**Comment by jkowens** [comment]
@leonidkroka I missed that that 4.2 specs had a failure. I'm ok with dropping support for AR 4.2 if necessary. Should we go that route or can that be resolved easily?

**Comment by leonidkroka** [comment]
@jkowens AR 4.2 has quite a different approach to building `columns_hash`
It's possible to go with if-else condition:
```
if respond_to?(:ignored_columns) && ignored_columns.any?
  connection.schema_cache.columns_hash(table_name)
else
  columns_hash
end
```
Have just opened the respective PR https://github.com/zdennis/activerecord-import/pull/800 (will rerun tests tomorrow)

Yeap, that doesn't look like a cool solution, so it's up to you to use that fix or drop support for AR 4.2

**Comment by jkowens** [comment]
Finally released in v1.5.0 🎉

File: lib/activerecord-import/import.rb [hunk 1]
```diff
@@ -961,6 +964,11 @@ def import_associations(models, options)
       end
     end
 
+    def schema_columns_hash
+      load_schema unless schema_loaded?
+      @schema_columns_hash ||= connection.schema_cache.columns_hash(table_name)
+    end
```
Comment (koshigoe) [inline lib/activerecord-import/import.rb:970]: @leonidkroka @jkowens 

I think the `@schema_columns_hash` should be cleared by `.reset_column_information`.

👉 https://github.com/zdennis/activerecord-import/issues/809

**Pull request merged**